grails-app/controllers/org/pih/warehouse/core/LocationController.groovy

```diff
@@ -82,6 +82,10 @@ class LocationController {
             if (selectedOrganization && (selectedOrganization?.id != currentOrganization?.id) && !selectedOrganization?.active) {
                 throw new IllegalArgumentException("The organization ${selectedOrganization?.name} is inactive, you can't assign it to the location")
             }
+            // If none supported activities are chosen, assign "None"
+            if (locationInstance?.id) {
```

Comment:
@jmiranda @awalkowiak this is the point I was trying to make on the tech huddle when I was saying 

> "I'm scared of how to distinguish whether I create or edit a location inside an update method"

because without this `if`, I'd assign `None` to the location while creating it from the scratch, not editing.

Fortunately it was as simple as adding this `if (locationInstance?.id)`

Reply:
Actually, the logic I'd like to see here would be something like this since it's more maintainable/readable. The same thing happens with your code but you need to think it through to see 
```
if (locationInstance?.id && params.list("supportedActivities").empty) {
    params.supportedActivities = [ActivityCode.NONE.id]
}
```

Please test to make sure empty works there under all scenarios 
* params.supportedActivities not present as a request parameter
* params.supportedActivities is present but set to null or empty (supportedActivities=null, supportedActivities=)
* params.supportedActivities is present and set to None (supportedActivities=NONE)
* params.supportedActivities is present and set to multiple activities

Reply:
It was not working for the case when the `params.supportedActivities = ""`, because it was creating: `[""]`, so it was not empty anymore. A workaround for this was to just make additional `.findAll{ it }` to filter out falsy values from the array, which `""` is.
If the `params.supportedActivities` is `null`, it was returning `[]`, so it's fine.
For the rest, so `params.supportedActivities = ["NONE"]` etc. it was obviously working fine.